pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -375,6 +380,47 @@ func (ks *scaler) scale(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscal
 		return desiredScale, nil
 	}
 
+	// Before we apply scale to zero and since we have failed to activate, check if any pod is in waiting state
+	// This should capture any case where we scale from zero and regular progressDeadline expiration will not be hit due to K8s limitations
+	// when not being in a deployment rollout.
+	if desiredScale == 0 && podCounter != nil {
+		ready, _, _, _, err := podCounter.PodCountsByState()
+		if err != nil {
+			return desiredScale, fmt.Errorf("error getting pod counts: %w", err)
+		}
+		if ready == 0 && currentScale > 0 {
+			var pod *corev1.Pod
+			pod, err = podCounter.GetAnyPod()
+			if err != nil {
+				return desiredScale, fmt.Errorf("error getting a pod: %w", err)
+			}
+			checkForPodErrorsBeforeScalingDown(logger, pod, pa)
+		}
+	}
+
 	logger.Infof("Scaling from %d to %d", currentScale, desiredScale)
 	return desiredScale, ks.applyScale(ctx, pa, desiredScale, ps)
 }
+
+func checkForPodErrorsBeforeScalingDown(logger *zap.SugaredLogger, pod *corev1.Pod, pa *autoscalingv1alpha1.PodAutoscaler) {
+	if pod != nil {
+		for _, status := range pod.Status.ContainerStatuses {
+			if status.Name != revresources.QueueContainerName {
+				if t := status.LastTerminationState.Terminated; t != nil {
+					logger.Debugf("marking exiting with: %d/%s", t.ExitCode, t.Message)
+					if t.ExitCode == 0 && t.Message == "" {
```

Comment:
Wouldn't this condition capture terminated pods that exited successfully? eg. normal scale down?